torch/csrc/distributed/rpc/init.cpp

```diff
@@ -220,8 +219,8 @@ PyObject* rpc_init(PyObject* /* unused */) {
               )")
           .def(
               "rpc_sync",
-              [&](PyRRef& self) {
-                return self.createRRefProxy(self, RRefProxyType::RPC_SYNC);
+              [](const PyRRef& self) {
```

Comment:
@mrshenli 
This PR introduces flakiness to RRefProxy tests
This is necessary to remove the flakiness. 

I don't know the exact reason yet. 
Tried this change and it works. 
It's inspired by my recent experience about how to fix pybind introduced flakiness.